File: app/dbmodels/SoftwareLicense.php
<?php

$table = array(
    'fields'        => array(),
    'oldfields'     => array(),
    'renamefields'  => array(),
    'indexes'       => array(),
    'oldindexes'    => array(),
    'relationships' => array(),
    'model'         => 'SoftwareLicense',
    'menu'          => ''
);
$table['fields'] = array(
    'id'                        => DBModels::type('increments',
                                                  array('visible' => false)),
    'name'                      => DBModels::type('string'),
    'comment'                   => DBModels::type('text'),
    'software_id'               => DBModels::type('integer'),
    'software_license_type_id'  => DBModels::type('integer'),
    'number'                    => DBModels::type('integer'),
    'serial'                    => DBModels::type('string'),
    'inventory_number'          => DBModels::type('string'),
    'software_version_buy_id'   => DBModels::type('integer'),
    'software_version_use_id'   => DBModels::type('integer'),
    'is_valid'                  => DBModels::type('boolean',
                                                  array('visible' => false)),
    'expire'                    => DBModels::type('date'),
);

$table['relationships'] = array(
    'assets'     => array(
        'type'  => 'belongsToMany',
        'item'  => 'Asset',
        'linktable' => 'used_software_licenses',
        'linkfields' => array('is_deleted', 'is_dynamic')),
    'software' => array(
        'type' => 'belongsTo',
        'item' => 'Software'),
    'software_version_buy' => array(
        'type' => 'belongsTo',
        'item' => 'SoftwareVersion'),
    'software_version_use' => array(
        'type' => 'belongsTo',
        'item' => 'SoftwareVersion'),
    'software_license_type' => array(
        'type' => 'belongsTo',
        'item' => 'SoftwareLicenseType'),
);

include "_commonentitylink.php";
include "_commonrecursiveentitylink.php";

File: fusionglpi/database/software_license.py
from fusionglpi import db

class SoftwareLicense(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    entity_id = db.Column(db.Integer, db.ForeignKey('entity.id'))
    is_recursive = db.Column(db.Boolean, nullable=False, server_default="0")
    name = db.Column(db.String(255, None, True))
    comment = db.Column(db.Text)
    software_id = db.Column(db.Integer, db.ForeignKey('software.id'))
    software_license_type_id = db.Column(db.Integer, db.ForeignKey('software_license_type.id'))
    number = db.Column(db.Integer, nullable=False, server_default="0")
    serial = db.Column(db.String(255, None, True))
    inventory_number = db.Column(db.String(255, None, True))
    software_version_buy_id = db.Column(db.Integer, db.ForeignKey('software_version.id'))
    software_version_use_id = db.Column(db.Integer, db.ForeignKey('software_version.id'))
    is_valid = db.Column(db.Boolean, nullable=False, server_default="0")
    expire = db.Column(db.Date)

    used_software_licenses = db.relationship('UsedSoftwareLicense', backref='software_license')
